src/backend/distributed/executor/multi_utility.c

```diff
@@ -3823,6 +3834,79 @@ PlanGrantStmt(GrantStmt *grantStmt)
 }
 
 
+/*
+ *  CollectGrantTableList determines and returns a list of table Oids from
+ *  grant statement.
+ *  Grant statement may appear in two form
+ *  1 - grant on table:
+ *      each distributed table oid in grant object list is added to returned list.
+ *  2 - grant all tables in schema:
+ *     Collect namespace oid list from grant statement
+ *     Add each distributed table oid in the target namespace list to the returned list.
+ */
+static List *
+CollectGrantTableList(GrantStmt *grantStmt)
+{
+	List *grantTableList = NIL;
+	ListCell *objectCell = NULL;
+	List *namespaceOidList = NIL;
+	bool grantOnTableCommand = false;
+	bool grantAllTablesOnSchenaCommand = false;
+
+
+	grantOnTableCommand = (grantStmt->targtype == ACL_TARGET_OBJECT &&
+						   grantStmt->objtype == RELATION_OBJECT_TYPE);
+	grantAllTablesOnSchenaCommand = (grantStmt->targtype == ACL_TARGET_ALL_IN_SCHEMA &&
+									 grantStmt->objtype == RELATION_OBJECT_TYPE);
+
+	/* we are only interested in table level grants */
+	if (!grantOnTableCommand && !grantAllTablesOnSchenaCommand)
+	{
+		return NIL;
+	}
+
+	if (grantAllTablesOnSchenaCommand)
+	{
+		List *distTableOidList = DistTableOidList();
+		ListCell *distributedTablOidCell = NULL;
+
+		foreach(objectCell, grantStmt->objects)
+		{
+			char *nspname = strVal(lfirst(objectCell));
+			bool missing_ok = false;
+			Oid nspId = get_namespace_oid(nspname, missing_ok);
+			Assert(nspId != InvalidOid);
+			namespaceOidList = list_append_unique_oid(namespaceOidList, nspId);
+		}
+
+		foreach(distributedTablOidCell, distTableOidList)
+		{
+			Oid relid = lfirst_oid(distributedTablOidCell);
+			Oid nspId = get_rel_namespace(relid);
```

Comment:
we should be using wordy variable names (e.g. `namespaceOid`)

Reply:
done